cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataSubscriberService.java

```diff
@@ -224,7 +224,7 @@ protected void processMessages(DatasetContext datasetContext,
           case ENTITY_CREATION:
           case ENTITY_DELETION:
             return new ProfileMetadataMessageProcessor(
-              datasetContext, datasetFramework, metadataStore, transactionRunner);
+              datasetContext, datasetFramework, metadataStore, transactionRunner, structuredTableContext);
```

Comment:
does this need both a structuredTableContext and a transactionRunner?

Reply:
Good point. The transactionRunner was added during the AppMetadataStore migration since the transactions were split. In this PR since I have consolidated the transaction at the MetadataSubscriberService, I can merge the transactions back and remove the transactionRunner from this call.